ros2topic/test/test_echo_pub.py

```diff
@@ -433,3 +433,24 @@ def publish_message():
             # Cleanup
             self.node.destroy_timer(publish_timer)
             self.node.destroy_publisher(publisher)
+
+    @launch_testing.markers.retry_on_failure(times=5)
+    def test_echo_timeout(self, launch_service, proc_info, proc_output):
+        topic = '/clitest/topic/echo_timeout'
+
+        command_action = ExecuteProcess(
+            cmd=['ros2', 'topic', 'echo', '--timeout', '5', topic, 'std_msgs/msg/String'],
```

Comment:
If we are just going to test that the timeout works, I'll suggest reducing this to 2 (or maybe even 1) seconds.  Just to keep CI times down.

Reply:
[5c620d6](https://github.com/ros2/ros2cli/pull/792/commits/5c620d6b3ae283e803ad9164ce5454397a412551)